scvi/models/modules.py

```diff
@@ -314,7 +314,7 @@ def forward(
         raw_px_scale = self.factor_regressor(z, *cat_list)
         px_scale = torch.softmax(raw_px_scale, dim=-1)
         px_dropout = self.px_dropout_decoder(z, *cat_list)
-        px_rate = torch.exp(library) * px_scale
+        px_rate = library + px_scale
```

Comment:
This is interesting. What are the actual units of library size here?

Reply:
Looks like the library size is already in log units

Reply:
indeed library in the scVI code is on the log scale

Reply:
you will need logsoftmax here for px_scale